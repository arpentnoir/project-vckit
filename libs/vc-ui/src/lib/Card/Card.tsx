import {
  Card as MUICard,
  CardActions,
  CardContent,
  CardProps,
  Stack,
} from '@mui/material';
import { Button, Text } from '..';

interface ICard extends CardProps {
  name: string;
  text: string;
  actionLabel?: string;
  handleAction?: () => void;
  headerAction?: React.ReactNode;
}

export const Card = ({
  name,
  text,
  handleAction,
  actionLabel,
  headerAction,
  ...rest
}: ICard) => {
  return (
    <MUICard sx={{ width: '320px' }} elevation={1} {...rest}>
      <CardContent>
        <Stack
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Text fontWeight="bold" variant="body1">
            {name}
          </Text>
          {headerAction}
        </Stack>

        <Text paddingTop="16px">{text}</Text>
      </CardContent>
      {actionLabel && handleAction && (
        <CardActions>
          <Button
            variant="text"
            color="error"
            label={actionLabel}
            onClick={handleAction}
            sx={{ paddingLeft: 0 }}
            textProps={{
              fontSize: 'small',
              fontWeight: 'bold',
              sx: { textDecoration: 'underline' },
            }}
          />
        </CardActions>
      )}
    </MUICard>
  );
};
