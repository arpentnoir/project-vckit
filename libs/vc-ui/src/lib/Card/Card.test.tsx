import { Switch } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import { Card } from './Card';

const mockHandleAction = jest.fn();

describe('Card', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    const { baseElement } = render(
      <Card
        name="Bob the Builder"
        text="Can we fix it?"
        handleAction={mockHandleAction}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should call correct function when item is clicked', () => {
    const { getByTestId } = render(
      <Card
        name="Bob the Builder"
        text="Can we fix it?"
        actionLabel="Remove"
        handleAction={mockHandleAction}
      />
    );

    fireEvent.click(getByTestId('button:Remove'));

    expect(mockHandleAction).toHaveBeenCalled();
  });

  it('should show header action if present', () => {
    const { getByTestId } = render(
      <Card
        name="Bob the Builder"
        text="Can we fix it?"
        headerAction={
          <Switch onClick={mockHandleAction} data-testid="switch" />
        }
      />
    );

    fireEvent.click(getByTestId('switch'));

    expect(mockHandleAction).toHaveBeenCalled();
  });
});
