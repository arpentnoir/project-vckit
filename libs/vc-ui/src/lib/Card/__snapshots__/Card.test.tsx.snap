// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-nblda9-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="css-3y1137-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
            tabindex="0"
          >
            Bob the Builder
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-ik0l5y-MuiTypography-root"
          tabindex="0"
        >
          Can we fix it?
        </p>
      </div>
    </div>
  </div>
</body>
`;
